Update radio_getradiosasync_548754145.md

HappyDawgie opened the pull request:
This problem (that is, design feature) was noticed in 2016 ( see https://github.com/Microsoft/cppwinrt/issues/47).

I verified that  Radios.GetAllAsync() works only when compiled for an x64 target.  It does not work for either x86 or AnyCpu.  With those two targets, a non-null unknown COM object is returned that will break the code and bypass a try-catch mechanism. The exception can be trapped with a try-finally.

HappyDawgie committed:
Update radio_getradiosasync_548754145.md

HappyDawgie commented:
I am unfamiliar with routing this serious defect to the Microsoft team involved... Can you assist in pointing the way?

mikematteson commented:
@laurenhughes : PR **passed** review checklist. Please evaluate proposed doc changes further, and note that the contributor has also asked a question. Thanks!

laurenhughes committed:
Updating app architecture info

Adding info provided in the pull request comment to the doc. Currently, this behavior is by-design and the documentation will be updated accordingly.

HappyDawgie commented:
I think this is also slightly incorrect, relating to apps compiled for 
"AnyCPU".  For whatever reason, this compilation option acts like an x86 
compilation option, and should be so noted.


On 12/11/17 11:35 AM, Keith Horton wrote:
>
> *@kehorton* requested changes on this pull request.
>
> This is slightly incorrect:
>
>   * When called from a UWP app, there is no architecture requirement.
>     It will work on the native architecture, as well as WOW64 (x86 on
>     x64 or ARM64).
>   * When called from a desktop Win32 app, this will only retrieve
>     radio instances when the app is the native architecture: WOW64
>     versions of the app (x86 on x64) will not see any radio instances.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/MicrosoftDocs/winrt-api/pull/235#pullrequestreview-82612123>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AQLeSTc4PEy7lKttMZjP6rch9LzpKLaqks5s_YQdgaJpZM4Q6G7X>.
>

hotlips commented:
Are there any alternative way to detect a generic USB BT radio-dongle device in Win32 than using GetRadiosAsync?
